ESF-905 Trustly documentation (#421)
SwedbankPay/developer.swedbankpay.com: payments/trustly/redirect.md

---
title: Swedbank Pay Trustly Payments
sidebar:
  navigation:
  - title: Trustly Payments
    items:
    - url: /payments/trustly
      title: Introduction
    - url: /payments/trustly/redirect
      title: Redirect
    - url: /payments/trustly/seamless-view
      title: Seamless View
    - url: /payments/trustly/after-payment
      title: After Payment
    - url: /payments/trustly/other-features
      title: Other Features
---

## Introduction

* When properly set up in your merchant/webshop site and the payer starts the
  purchase process, you need to make a `POST` request towards Swedbank Pay with

>   your Purchase information. This will generate a payment object with a unique

Reviewer: ```suggestion
  your Purchase information. This will generate a `payment` resource with a unique
```

>   `paymentID`. You will receive a **redirect URL** to a Swedbank Pay payment

Reviewer: ```suggestion
  `id`. You will receive a **redirect URL** to a Swedbank Pay payment
```

  page (`redirect-sale` operation).
* You need to redirect the payer's browser to that specified URL so that the
  payer can enter the payment details in a secure Swedbank Pay environment.
* Swedbank Pay will redirect the payer's browser to - one of two specified URLs,
  depending on whether the payment session is followed through completely or
  cancelled beforehand. Please note that both a successful and rejected payment
  reach completion, in contrast to a cancelled payment.
* When you detect that the payer reach your `completeUrl` , you need to do a
  `GET` request to receive the state of the transaction, containing the

>   `paymentID` generated in the first step, to receive the state of the

Reviewer: ```suggestion
  `id` URI generated in the first step, to receive the state of the
```

        },
        "metadata": {
            "id": "/psp/trustly/payments/{{ page.payment_id }}/metadata"
        }
    },
    "operations": [
        {
            "method": "PATCH",
            "href": "{{ page.api_url }}/psp/trustly/payments/{{ page.payment_id }}",
            "rel": "update-payment-abort"
        },
        {
            "method": "POST",
            "href": "{{ page.api_url }}/psp/trustly/payments/{{ page.payment_id }}/sales",
            "rel": "create-sale"
        },
        {
            "method": "GET",
            "href": "https://ecom.externalintegration.payex.com/trustly/payments/sales/8f3ba6c8f4e3f6125ae6c18bec15c612747cf2c35dc5cac35d4bebc10cf7317e",
            "rel": "redirect-sale"
        }
    ]
}
```

## Trustly flow

The sequence diagram below shows the two requests you have to send to Swedbank
Pay to make a purchase.
The diagram also shows the process of a complete purchase in high level.

```mermaid
sequenceDiagram
    Consumer->>Merchant: Start purchase
    activate Merchant
    note left of Merchant: First API request
    Merchant->>-Swedbank Pay: POST <Trustly Payment> (operation=Purchase)
    activate Swedbank Pay

>     Swedbank Pay-->>-Merchant: payment resource

Reviewer: Use code-like names for the identifiers (not textual descriptions) within the diagram and define an alias for it with `participant SwedbankPay as Swedbank Pay` at the top instead.

```suggestion
    SwedbankPay-->>-Merchant: payment resource
```

---

> title: Swedbank Pay Trustly Payments Redirect

Reviewer: ```suggestion
title: Swedbank Pay Trustly Payments – Redirect
```
